### PR TITLE
Soft delete and update project

### PR DESCRIPTION
**Pull request by Georgeygigz**
- implementation to update project
- implementation to delete project
- implement soft delete

### CODE REVIEW

**Review by kwameAsiago — APPROVED** [review]

**Review by Ruiru11 — APPROVED** [review]
This looks good but its advisable to do a mongo soft delete maybe using `mongoose_delete` that creates fields on the object model in the db directly without you having to find and update. But this is good stuff

**Review by neymarjimoh — COMMENTED** [review]
this is nice. there should also be room for deleting projects from the database permanently

**Comment by Nicanor008** [comment]
> This looks good but its advisable to do a mongo soft delete maybe using `mongoose_delete` that creates fields on the object model in the db directly without you having to find and update. But this is good stuff

good idea, I didn't know there's a tool to handle that. I've created this issue [here](https://github.com/Nicanor008/shoman-api/issues/15). But not urgently needed on MVP release

**Comment by Nicanor008** [comment]
> this is nice. there should also be room for deleting projects from the database permanently

yes, I agree with you @neymarjimoh, however, for MVP release we will require all projects. if deleted, Shoman admin is granted access and the admin can decide what to do with the project on update. Permanent deletion can only be done by admin

File: src/api/v1/project/project_controllers.js [hunk 1]
```diff
@@ -45,15 +47,78 @@ export const GetProjectsController = (req, res, next) => {
...
+                message: 'Error when updating project with id=' + projectId,
+            })
+        })
+}
+
+export const DeleteProjectController = (req, res) => {
+    const { projectId } = req.params
+    const query = { _id: new mongodb.ObjectId(projectId), isDeleted: false }
+
+    Project.findOneAndUpdate(query, { isDeleted: true })
```
Comment (Nicanor008) [inline src/api/v1/project/project_controllers.js:108]: grant showman access after deleting, `showmanAccess: true`
```suggestion
    Project.findOneAndUpdate(query, { isDeleted: true, showmanAccess: true })
```

**Review by Nicanor008 — CHANGES_REQUESTED** [review]

**Comment by neymarjimoh** [comment]
> > this is nice. there should also be room for deleting projects from the database permanently
> 
> yes, I agree with you @neymarjimoh, however, for MVP release we will require all projects. if deleted, Shoman admin is granted access and the admin can decide what to do with the project on update. Permanent deletion can only be done by admin

Alright @Nicanor008

**Review by Nicanor008 — APPROVED** [review]

**Pull request merged**